app/controllers/join_team_requests_controller.rb

```diff
@@ -3,40 +3,38 @@ class JoinTeamRequestsController < ApplicationController
   # GET /join_team_requests.xml
 
   def action_allowed?
-    current_role_name.eql?("Student")
+    ['Student', 'Instructor', 'Teaching Assistant'].include?(current_role_name) #people with this roles can only access the function provied by the controller 
   end
 
-  def index
-    @join_team_requests = JoinTeamRequest.all
-
-    respond_to do |format|
-      format.html # index.html.erb
-      format.xml  { render :xml => @join_team_requests }
+  private 
+  def render_request
+      respond_to do |format|
+      format.html 
+      format.xml  { render :xml => @join_team_request } #displays the join team instance
     end
   end
 
+  def index
```

Comment:
Do we have a index page to for join_team_requests? Why do a JoinTeamRequest.all?